school_center.py

```diff
@@ -52,6 +55,17 @@ def sort_key(c):
     within_distance = []
     nearest_distance = None;
     nearest_center = None
+
+     # Use distance cache
+    cache_key = (school_lat, school_long)
```

Comment:
distance is a function of two sets of co-ordinates ie four float values. you are using only two values as cache key, dont think this will work